modules/apps/29-fee/ibc_module.go

```diff
@@ -175,8 +175,19 @@ func (im IBCModule) OnRecvPacket(
 	ctx sdk.Context,
 	packet channeltypes.Packet,
 	relayer sdk.AccAddress,
-) ibcexported.Acknowledgement {
-	// TODO: Implement fee specific logic if fee is enabled for the given channel
+) exported.Acknowledgement {
+	if im.keeper.IsFeeEnabled(ctx, packet.SourcePort, packet.SourceChannel) {
+		ack := im.app.OnRecvPacket(ctx, packet, relayer)
+		forwardRelayer, found := im.keeper.GetCounterpartyAddress(ctx, relayer.String())
+		if !found {
+			forwardRelayer = ""
+		}
+
+		return types.IncentivizedAcknowledgement{
+			Result:                ack.Acknowledgement(),
+			ForwardRelayerAddress: forwardRelayer,
+		}
+	}
 	return im.app.OnRecvPacket(ctx, packet, relayer)
```

Comment:
```suggestion
	}
	
	return im.app.OnRecvPacket(ctx, packet, relayer)
```

super small nit/suggestion. Another idea is to move ics29 logic into `keeper/`. Or adjust the code such that:
```go
    if !im.keeper.IsFeeEnabled() {
        return im.app.OnRecvPacket
    }
    
    // do ics29 logic
``` 

Mostly just looking to make sure the no-op is very clear when the fee is not enabled, how this is done is up to you. I don't think there's a "best" way